test/src/androidTest/kotlin/io/realm/shared/RealmObjectTests.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.shared
+
+import io.realm.Realm
+import io.realm.RealmConfiguration
+import io.realm.VersionId
+import io.realm.util.PlatformUtils
+import io.realm.version
+import test.link.Child
+import test.link.Parent
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+class RealmObjectTests {
+
+    companion object {
+        // Expected version after writing Parent to Realm
+        private val EXPECTED_VERSION = VersionId(3, 2)
```

Comment:
#magicalnumber. Just out of curiosity: Is there any straight forward way to reason why this is the right number?

Reply:
The Realm Version is `(2,1)` when we open a normal Realm (due to us writing the schema), so it becomes version 3 once we write an object. The `index` is some internal accounting used by Core (at least that was the explanation I got).

I have been considering if we should just hide the index since it isn't used when comparing versions anyway, but I wasn't sure if it would be relevant for debugging. 

Do you have any strong opinions either way?

Reply:
It would mean it couldn't be a data class, but that wouldn't be a big issue

Reply:
Taking to Finn, the `index` is an index into an internal ringbuffer in Core, so it doesn't have any practical use. I'll remove it.